x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -411,12 +403,23 @@ private static Tuple<Mode, String> parsePolicyName(Token policyToken) {
 
     @Override
     public PlanFactory visitLookupCommand(EsqlBaseParser.LookupCommandContext ctx) {
-        return p -> {
-            var source = source(ctx);
-            UnresolvedAttribute tableName = new UnresolvedAttribute(source(ctx.tableName), ctx.tableName.getText());
-            List<Attribute> matchFields = visitQualifiedNames(ctx.matchFields);
-            return new Lookup(source, p, tableName, matchFields, null, null);
-        };
+        var source = source(ctx);
+
+        List<NamedExpression> matchFields = visitQualifiedNamePatterns(ctx.qualifiedNamePatterns(), ne -> {
+            if (ne instanceof UnresolvedNamePattern || ne instanceof UnresolvedStar) {
+                var src = ne.source();
+                throw new ParsingException(src, "Using wildcards [*] in LOOKUP ON is not allowed yet [{}]", src.text());
+            }
+        });
+
+        Literal tableName = new Literal(source, visitQualifiedNamePattern(ctx.tableName, ne -> {
+            if (ne instanceof UnresolvedNamePattern || ne instanceof UnresolvedStar) {
```

Comment:
It's so `instanceof`! I wish there were some named way of saying "precise" or something. That can be a problem for another time though.